### ISSUE TITLE
The demo/pwm_ats application can't fade LED

### ISSUE THREAD

**Issue by master-q** [fpiot/arduino-ats#4]
Is there problem at C-language side or ATS-language side?

**Comment by master-q** [comment]
I think it's caused by C-language. Also following code doesn't fade LED.

```ats
#include "config.hats"
#include "{$TOP}/avr_prelude/kernel_staload.hats"

staload "{$TOP}/SATS/arduino.sats"

#define BLINK_DELAY_MS 500.0

%{
int brightness = 0;
int fadeAmount = 5;

void loop_c(void)  {
  analogWrite(9, brightness);
  brightness = brightness + fadeAmount;
  if (brightness == 0 || brightness == 250) {
    fadeAmount = -fadeAmount ;
  }
  _delay_ms(30.0);
}
%}

extern fun loop_c (): void = "mac#"

implement main0 () = {
  fun loop () = {
    val () = loop_c ()
    val () = loop ()
  }
  val () = pinMode (LED_BUILTIN, OUTPUT)
  val () = loop ()
}
```

**Comment by master-q** [comment]
Following switch on 9th LED.

```c
analogWrite(9, 255);
```

And following switch off the LED.

```c
int i;
for (i = 0; i < 255; i++) {
  analogWrite(9, i);
    printf("i = %d\n", i);
}
```

**Comment by master-q** [comment]
Fixed at e389fd03d1f5440eefde2d4984a5d7524bff8e81.
